File: SubmitInstance/InstancePoster.cs
using SimpleSOAPClient;
using SimpleSOAPClient.Helpers;
using SimpleSOAPClient.Models;
using System.Xml.Linq;

namespace SubmitInstance
{
	internal class InstancePoster : IInstancePoster
	{
		public async Task<XDocument> Post(string server, string action, XDocument body)
		{
			SoapClient client = SoapClient.Prepare();
			SoapEnvelope requestEnvelope = SoapEnvelope.Prepare().Body(body);
			SoapEnvelope responseEnvelope = await client.SendAsync(server, action, requestEnvelope);
			return new XDocument(responseEnvelope.Body.Value);
		}
	}
}

File: VB/SubmitInstance/InstancePoster.vb
Imports SimpleSOAPClient
Imports SimpleSOAPClient.Helpers
Imports SimpleSOAPClient.Models

Friend Class InstancePoster
	Implements IInstancePoster

	Public Async Function Post(ByVal server As String, ByVal action As String, ByVal body As XDocument) As Task(Of XDocument) Implements IInstancePoster.Post
		Dim client As SoapClient = SoapClient.Prepare()
		Dim requestEnvelope As SoapEnvelope = SoapEnvelope.Prepare().Body(Of XElement)(body.Root)
		Dim responseEnvelope As SoapEnvelope = Await client.SendAsync(server, action, requestEnvelope)
		Return New XDocument(responseEnvelope.Body.Value)
	End Function
End Class